src/app/features/automated-status-update/automated-status-update-config-service.ts

```diff
@@ -0,0 +1,342 @@
+      rule: any;
+    }[]
+  >();
+
+  constructor(
+    private entityRegistry: EntityRegistry,
+    private entityMapper: EntityMapperService,
+    private dialog: MatDialog,
+    private unsavedChangesService: UnsavedChangesService,
+    private entitySchemaService: EntitySchemaService,
+    private configService: ConfigService,
+    private entityRelationsService: EntityRelationsService,
+  ) {
+    this.configService.configUpdates.subscribe(() =>
+      // wait until EntityConfigService has updated the entityRegistry
+      setTimeout(() => this.buildDependencyMap()),
+    );
+  }
+
+  /**
+   * Builds a map of dependent field rules from entity schema config
+   * to be used during automatic updates.
+   */
+  private buildDependencyMap() {
+    this.dependencyMap.clear();
+    for (const targetType of this.entityRegistry.values()) {
+      for (const [fieldKey, fieldConfig] of targetType.schema.entries()) {
+        this.processFieldConfig(targetType, fieldKey, fieldConfig);
+      }
+    }
+  }
+
+  /**
+   * Processes a single field configuration to extract automated rules
+   */
+  private processFieldConfig(
+    targetType: EntityConstructor,
+    fieldKey: string,
+    fieldConfig: EntitySchemaField,
+  ) {
+    const rule = fieldConfig.defaultValue?.config;
+    if (
+      fieldConfig.defaultValue?.mode === "updated-from-referencing-entity" &&
+      rule
+    ) {
+      this.processAutomatedRule(targetType, fieldKey, rule);
+    }
+  }
+
+  /**
+   * Processes a single automation rule and adds it to the dependency map
+   * @param targetEntityType - Type of entity that contains the rule
+   * @param targetFieldId - Field ID where the rule is defined
+   * @param rule - Automation rule configuration
+   */
+  private processAutomatedRule(
+    targetEntityType: EntityConstructor,
+    targetFieldId: string,
+    rule: any,
+  ) {
+    const key = `${rule.relatedEntityType}|${rule.relatedTriggerField}`;
+    const entry = {
+      targetEntityType: targetEntityType,
+      targetFieldId: targetFieldId,
+      relatedReferenceField: rule.relatedReferenceField,
+      rule: rule,
+    };
+
+    const existingEntries = this.dependencyMap.get(key) || [];
+    this.dependencyMap.set(key, [...existingEntries, entry]);
+  }
+
+  /**
+   * Returns a list of related entities with reference info for the given entity type.
+   * Used in automated rule configuration.
+   */
+  public updateAvailableRelatedEntityForAutomated(entityType: string): {
+    label: string;
+    entityType: string;
+    relatedReferenceFields: string[];
+  }[] {
+    const relatedEntities =
+      this.entityRelationsService.getEntityTypesReferencingType(entityType);
+    return relatedEntities
+      .filter((refType) => !!refType.entityType.label)
+      .map((refType) => ({
+        label: refType.entityType.label,
+        entityType: refType.entityType.ENTITY_TYPE,
+        relatedReferenceFields: refType.referencingProperties.map((p) => p.id),
+      }));
+  }
```

Comment:
if this doesn't depend on anything other in this service, maybe we should move it to the component itself, now that the admin components or smaller.